ansible/roles/vm_set/library/vm_topology.py

```diff
@@ -322,11 +328,24 @@ def add_injected_fp_ports_to_docker(self):
             PTF (int_if) ----------- injected port (ext_if)
 
         """
-        for vlan in self.injected_fp_ports:
-            (_, _, ptf_index) = VMTopology.parse_vm_vlan_port(vlan)
-            ext_if = adaptive_name(INJECTED_INTERFACES_TEMPLATE, self.vm_set_name, ptf_index)
-            int_if = PTF_FP_IFACE_TEMPLATE % ptf_index
-            self.add_veth_if_to_docker(ext_if, int_if)
+        for vm, vlans in self.injected_fp_ports.items():
+            for vlan in vlans:
+                (_, _, ptf_index) = VMTopology.parse_vm_vlan_port(vlan)
+                ext_if = adaptive_name(INJECTED_INTERFACES_TEMPLATE, self.vm_set_name, ptf_index)
+                int_if = PTF_FP_IFACE_TEMPLATE % ptf_index
+                properties = self.vm_properties.get(vm, {})
+                create_vlan_subintf = properties.get('device_type') in BACKEND_DEVICE_TYPES
```

Comment:
shouldn't this be 'dut_type'?

Reply:
There is a property called `device_type` defined in the VM properties: https://github.com/Azure/sonic-mgmt/blob/1be712a373688b8ff974d9ccf7471d09bf33bc6b/ansible/vars/topo_t1-backend.yml#L68